pkg/service/presentation/service.go

```diff
@@ -308,3 +295,88 @@ func (s Service) ListDefinitions(ctx context.Context) (*model.ListDefinitionsRes
 
 	return resp, nil
 }
+
+func (s Service) CreateRequest(ctx context.Context, req model.CreateRequestRequest) (*model.Request, error) {
+	if err := sdkutil.IsValidStruct(req); err != nil {
+		return nil, err
+	}
+
+	request := req.PresentationRequest
+	pd, err := s.storage.GetDefinition(ctx, request.PresentationDefinitionID)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting presentation definition")
+	}
+	if pd == nil {
+		return nil, errors.Errorf("presentation definition %q is nil", request.PresentationDefinitionID)
+	}
+
+	requestID := uuid.NewString()
+	token, err := jwt.NewBuilder().Claim("presentation_definition", pd.PresentationDefinition).
```

Comment:
we should use https://github.com/TBD54566975/ssi-service/blob/main/internal/keyaccess/jwt.go

if we don't have exactly the right method - let's add it

Reply:
I disagree. That package doesn't have access to the `keystore`, and I don't think it should.

The `Sign` method of the `keystore` is calling keyaccess under the hood.

Reply:
it doesn't need to - here is how it's used https://github.com/TBD54566975/ssi-service/blob/12083e75bcac0d8e18b03bd4d7957fe83873a964/pkg/service/credential/service.go#L345

I'd like to stay consistent

Reply:
I'm not following the suggestion you're making. Would you mind clarifying what pieces to move where?

Reply:
I would remove the jwt builder in favor of some code like

```go
gotKey, err := s.keyStore.GetKey(ctx, keystore.GetKeyRequest{ID: request.IssuerKID})
if err != nil {
	return nil, sdkutil.LoggingErrorMsgf(err, "getting key for signing request<%s>", request.IssuerKID)
}
keyAccess, err := keyaccess.NewJWKKeyAccess(request.IssuerKID, gotKey.ID, gotKey.Key)
	if err != nil {
		return nil, errors.Wrapf(err, "creating key access for signing request with key<%s>", gotKey.ID)
	}
requestToken, err := keyAccess.Sign(request)
if err != nil {
	return nil, errors.Wrapf(err, "could not sign request with key<%s>", gotKey.ID)
}
```